pkg/proxy/userspace/roundrobin.go

```diff
@@ -226,9 +226,8 @@ func (lb *LoadBalancerRR) OnEndpointsAdd(endpoints *v1.Endpoints) {
 	lb.lock.Lock()
 	defer lb.lock.Unlock()
 
-	for portname := range portsToEndpoints {
-		svcPort := proxy.ServicePortName{NamespacedName: types.NamespacedName{Namespace: endpoints.Namespace, Name: endpoints.Name}, Port: portname}
-		newEndpoints := portsToEndpoints[portname]
+	for portName, newEndpoints := range portsToEndpoints {
```

Comment:
what was the reason for this change? I'm pretty sure the previous assignment was to avoid reuse of the iterator value because of https://github.com/golang/go/wiki/CommonMistakes#using-reference-to-loop-iterator-variable

Reply:
`newEndpoints` is a list and `util.ShuffleStrings` would make a copy of it.

Reply:
I'd still prefer to limit the changes in this PR to ones fixing the flake, especially if we intend to merge it in 1.19

Reply:
That's okay.